git_machete/git_operations.py

```diff
@@ -405,15 +413,20 @@ def create_branch(self, branch: str, out_of_revision: str) -> None:
         self.flush_caches()  # the repository state has changed because of a successful branch creation, let's defensively flush all the caches
```

Comment:
One nuance... `checkout` AFAICS can technically invalidate `__reflogs_cached` and all the `__..._by_revision_cached`... but under the single key,  `HEAD` 🤔

Reply:
The question is... are any of this caches used for the `HEAD` key?

Reply:
Even if they aren't currently... there's currently nothing in place that would forbid such usage

Reply:
> One nuance... `checkout` AFAICS can technically invalidate `__reflogs_cached` and all the `__..._by_revision_cached`... but under the single key, `HEAD` 🤔

How can it invalidate that cache?

Reply:
Just `flush_caches()` as well, let's not over-optimize with just deleting a single key

Reply:
Ok, I've added a `self.flush_caches` at the end of `GitContext.checkout` method